src/main.rs

```diff
@@ -133,15 +111,26 @@ fn create_message(repo: &Repository, matches: &clap::ArgMatches) -> i32 {
 ///
 fn find_tree_init_hash(repo: &Repository, matches: &clap::ArgMatches) -> i32 {
     // note: commit is always present since it is a required parameter
-    repo.revparse_single(matches.value_of("commit").unwrap())
-        .and_then(|obj| repo.find_commit(obj.id()))
-        .chain_err(|| EK::WrappedGitError)
+    repo.value_to_commit(matches.value_of("commit").unwrap())
         .and_then(|commit| repo.find_tree_init(commit).chain_err(|| EK::WrappedGitDitError))
         .map(|commit| {println!("{}", commit.id()); 0})
         .unwrap_or_else(|err| {error!("{}", err); 1})
 }
 
 
+/// get-issue-metadata subcommand implementation
+///
+fn get_issue_metadata(repo: &Repository, matches: &clap::ArgMatches) -> i32 {
+    // note: "head" is always present since it is a required parameter
+    let commits = try_or_1!(repo.value_to_commit(matches.value_of("head").unwrap())
+                                .map(|commit| IssueMessagesIter::new(commit, repo)));
+    for trailer in commits.flat_map(|commit| CommitExt::trailers(&commit)) {
```

Comment:
`flat_map(CommitExt::trailers)` doesn't work because we pass a ref. I guess I expanded it to test whether that was the only problem and forgot to write it as `commit.trailers()`.